table-project/src/components/UserTable.tsx

```diff
@@ -14,38 +38,54 @@ export default function UserTable() {
     staleTime: 10000,
   });
 
+  const rerender = useReducer(() => ({}), {})[1];
+
+  const options = {
+    data: data??[],
+    columns: columns,
+    getCoreRowModel: getCoreRowModel(),
+  }
+
+  const table = useReactTable(options);
+
   return (
-    <table>
-        <thead>
-            <tr>
-                <th>Name</th>
-                <th>Website</th>
-            </tr>
-        </thead>
-        {isLoading? (
-            <tbody>
-                <tr>
-                    <td>...</td>
-                    <td>...</td>
-                </tr>
-            </tbody>
-            ) : isError? (
-            <tbody>
-                <tr>
-                    <td>no users...</td>
-                    <td>...</td>
+    <div>
+        {(isLoading || isError)? (<div>...</div>):(
+        <>
+        <table>
+            <thead>
+            {table.getHeaderGroups().map(headerGroup => (
+                <tr key={headerGroup.id}>
+                {headerGroup.headers.map(header => (
+                    <th key={header.id}>
+                    {header.isPlaceholder
+                        ? null
+                        : flexRender(
+                            header.column.columnDef.header,
+                            header.getContext()
+                        )}
+                    </th>
+                ))}
                 </tr>
-            </tbody>
-            ):(
+            ))}
+            </thead>
             <tbody>
-                {data.map((user) => (
-                <tr key={user.id}>
-                    <td>{user.name}</td>
-                    <td>{user.website}</td>
-                </tr>
+            {table.getRowModel().rows.map(row => (
+                <tr key={row.id}>
+                {row.getVisibleCells().map(cell => (
+                    <td key={cell.id}>
+                    {flexRender(cell.column.columnDef.cell, cell.getContext())}
+                    </td>
                 ))}
+                </tr>
+            ))}
             </tbody>
```

Comment:
This doesn't look pretty, but it's _relatively_ easy to understand. We front-loaded all of the work to define whether columns should be searchable, what columns should look like, etc. 

Same table that does nothing, but we leave the door open for all of the complex state changes we're going to need.